blockdevice/stats.go

```diff
@@ -220,3 +310,190 @@ func (fs FS) SysBlockDeviceStat(device string) (IOStats, int, error) {
 	}
 	return stat, count, err
 }
+
+func (fs FS) SysBlockDeviceQueueStats(device string) (BlockQueueStats, error) {
+	stat := BlockQueueStats{}
+	addRandom, err := util.ReadUintFromFile(fs.sys.Path(sysBlockPath, device, sysBlockQueue, "add_random"))
```

Comment:
What about making the stat struct members all pointers and then use a loop like:
```
for file, val := range map[string]*uint64 {
  "add_random": stat.AddRandom,
  ...
} {
  val, err := util.ReadUintFromFile(fs.sys.Path(sysBlockPath, device, sysBlockQueue, file)
  ...
```

Reply:
not all fields are uint64, two of them are int64 and another two are strings and one is a *uint64, but it possible to factor all unit64 in such way and handle others as special cases.